app/Services/ModuleService.php

```diff
@@ -606,15 +607,44 @@ public function setupLanguages(): Collection
     }
 
     /**
-     * Load a module in a static scope, to prevent it from modifying local or object variables.
+     * Load a custom module in a static scope, to prevent it from modifying local or object variables.
      *
      * @param string $filename
      *
      * @return mixed
      */
     private static function load(string $filename)
     {
-        return include $filename;
+        try {
+            return include $filename;
+        } catch (Exception $loading_exception) {
+            self::disable_module($filename);
```

Comment:
We don't need to disable the modue (by renaming it).
It did not load, and is therefore already disabled.

I'm sure module developers will get very annoyed if the code renames their module every time they make a coding error or typo.

Instead, we could simply display the error.

```
FlashMessages::addMessage(e($ex->getMessage()), 'danger');
```

I don't think the full stack-dump is needed.  Errors here are likely to be class/trait/mismatches and will be clear from the error message.

Reply:
Sure, I'd put this in:
```
FlashMessages::addMessage(I18N::translate('Module %s errored while loading.', $module_name), 'danger');
FlashMessages::addMessage($loading_exception->getMessage(), 'danger');
```

… or really just the message?

Reply:
> really just the message

In reality, I would add a message telling the user that this module isn't compatible with webtrees, and that they should upgrade/delete it.

But I reviewed this over my morning coffee, and that was all I had time to type.... ;-)